packages/vitest/src/node/workspace.ts

```diff
@@ -323,10 +323,14 @@ export class WorkspaceProject {
   getSerializableConfig() {
     const optimizer = this.config.deps?.optimizer
     const poolOptions = this.config.poolOptions
+    const coverage = this.ctx.config.coverage
 
     // Resolve from server.config to avoid comparing against default value
     const isolate = this.server?.config?.test?.isolate
 
+    if (coverage.provider === 'v8' && coverage.enabled)
+      throw new Error('@vitest/coverage-v8 does not work with --browser. Use @vitest/coverage-istanbul instead')
```

Comment:
@sheremet-va can you think of any other place where this check could be done? Maybe a bit earlier than this one.